### PR TITLE
[tr] Add website config for Turkish

### PR DESCRIPTION
**Pull request by aliok**
### Describe your changes
- Added a section in `config.toml` for Turkish
- Added a `i18n/tr.toml` file for the common UI text translation

Google Docs for reference:
- [`i18n/tr.toml`](https://docs.google.com/document/d/1iiKTIeUz1UZouRIiLZfE6CFBGorXV6elcb-O5fkYNQ0/edit)
- [`config.toml`](https://docs.google.com/document/d/10i4QtWh_OMBkDHs0CI4JT82mLzOT1PTraks_IXoetHc/edit)

Preview: https://deploy-preview-2235--cncfglossary.netlify.app/tr/
(all empty right now)

### Checklist before opening this PR (put `x` in the checkboxes)
- [x] This PR does not contain plagiarism
  - don’t copy other people’s work unless you are quoting and contributing it to them.
- [x] I have signed off on all commits 
  - [signing off](https://docs.github.com/en/authentication/managing-commit-signature-verification/signing-commits) (ex: `git commit -s`) is to affirm that commits comply [DCO](https://wiki.linuxfoundation.org/dco). If you are working locally, you could add an alias to your `gitconfig` by running `git config --global alias.ci "commit -s"`.

### CODE REVIEW

**Comment by aliok** [comment]
cc @halil-bugol @eminalemdar @developer-guy @symys @rwxdash

**Review by halil-bugol — APPROVED** [review]
Approved

**Comment by aliok** [comment]
Converted to a draft to hold this PR. We're running a poll for a possible better translation of a term.

**Comment by aliok** [comment]
Changed one word as we voted. Ready for review!

**Review by eminalemdar — APPROVED** [review]
LGTM

**Review by rwxdash — APPROVED** [review]

**Comment by aliok** [comment]
Hm I thought we (approvers) would be able to merge to this branch ourselves @seokho-son . Do we need your approval?

**Comment by seokho-son** [comment]
@aliok 
localization approvers can merge PRs to their dev branch, but it is not about codeowners. 
Changing files outside /content/tr/* still requires approval from maintainers (or English approvers) even for a localization dev branch. Localization approvers probably don't want to deal with the tricky aspects of repo such as changing site configuration. ;)

File: config.toml
Comment (seokho-son) [inline config.toml:136]:
```suggestion
languageName ="Türkçe (Turkish)"
```
Is this suggestion acceptable? :)
(for consistency with other language scripts)

**Review by seokho-son — APPROVED** [review]
minor comment. :) 
the other parts looks good to me!

**Comment by seokho-son** [comment]
Please utilize the 'Squash and merge' method when merging a localization pull request (PR). While not mandatory, we strongly encourage you to make an effort to keep the commit history clean. In case the commit history in the development branch is not well-maintained, the maintainers would squash the commits into a single one when merging the development branch into the main branch. :)

**Comment by aliok** [comment]
> files outside /content/tr/* still requires approval from maintainers

Got it. Thanks

**Pull request merged**
